test/smoke/src/sql/areas/notebook/notebook.test.ts

```diff
@@ -22,7 +21,25 @@ export function setup() {
 
 			await app.workbench.sqlNotebook.runActiveCell();
 			await app.workbench.sqlNotebook.waitForResults();
+		});
+
+		it('can open ipynb file, run all, and save notebook with outputs', async function () {
+			const app = this.app as Application;
+			await app.workbench.sqlNotebook.openFile('hello.ipynb');
+			await app.workbench.sqlNotebook.waitForKernel('Python 3');
+
+			await app.workbench.sqlNotebook.clearResults();
+			await app.workbench.sqlNotebook.waitForAllResultsGone([1, 2, 3]);
+			await app.workbench.sqlNotebook.runAllCells();
+			await app.workbench.sqlNotebook.waitForAllResults([1, 2, 3]);
+
+			await app.workbench.quickaccess.runCommand('workbench.action.files.save');
 			await app.workbench.quickaccess.runCommand('workbench.action.closeActiveEditor');
+
+			await app.workbench.sqlNotebook.openFile('hello.ipynb');
+			await app.workbench.sqlNotebook.waitForKernel('Python 3');
+			// The cell ids increase by one after closing and reopening notebook
+			await app.workbench.sqlNotebook.waitForAllResults([4, 5, 6]);
```

Comment:
Looks great! I'm just curious about this line, do cell ids always increase by one after closing and reopening?

Reply:
I think the cell ids are dependent on how many CellModel objects are created after ADS is opened. So the first cell created in the previous test has id = 0, the first time hello.ipynb is opened, 3 CellModels are created with ids 1,2,3, and when hello.ipynb is re-opened, another 3 CellModels are created with ids 4,5,6.

Reply:
Do we start a new instance of ADS for every smoke test ran? if not what happens if we add another smoke test that opens a notebook and add cells - this is going to break at that point then right?

Instead what if you took in an index and then had the function grab the ordered list of cell elements and then used the passed in indexes to pick which ones to wait for?